projects/ancestor/ancestor.py

```diff
@@ -0,0 +1,78 @@
+# Definition for a binary tree node.
+
+class Stack():
+    def __init__(self):
+        self.stack = []
+    def push(self, value):
+        self.stack.append(value)
+    def pop(self):
+        if self.size() > 0:
+            return self.stack.pop()
+        else:
+            return None
+    def size(self):
+        return len(self.stack)
+
+class Graph:
+    """Represent a graph as a dictionary of vertices mapping labels to edges."""
+    def __init__(self):
+        self.vertices = {}
+
+    def add_vertex(self, vertex):
+        """
+        Add a vertex to the graph.
+        """
+        self.vertices[vertex] = set()
+
+    def add_edge(self, v1, v2):
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v2].add(v1)
+        else:
+            raise IndexError("That vertex doesn't exist")
+
+    def earliest_ancestor(self, start):
+        stack = Stack()
+        stack.push([start])
+        visited = set()
+        new_path = None
+        while stack.size() > 0:
+            path = stack.pop()
+            node = path[-1]
+            if node not in visited:
+                visited.add(node)
+                for next_node in self.vertices[node]:
+                    new_path = path.copy()
+                    new_path.append(next_node)
+                    stack.push(new_path)
+        
+        return new_path[-1]
```

Comment:
I'm not sure if you've done this already but I would suggest taking earliest ancestor out of the graph class. I'd also recommend using a queue. I tried working through yours for a little while but I only got about half way through passing the tests.